include/deal.II/base/quadrature.h

```diff
@@ -241,18 +241,116 @@ class Quadrature : public Subscriptor
   void
   serialize(Archive &ar, const unsigned int version);
 
+protected:
   /**
-   * This function returns true if the quadrature object is a tensor product
-   * of one-dimensional formulas and the quadrature points are sorted
-   * lexicographically.
+   * List of quadrature points. To be filled by the constructors of derived
+   * classes.
    */
-  bool
-  is_tensor_product() const;
+  std::vector<Point<dim>> quadrature_points;
+
+  /**
+   * List of weights of the quadrature points.  To be filled by the
+   * constructors of derived classes.
+   */
+  std::vector<double> weights;
+};
+
+
+/**
+ * This function returns true if the incoming Quadrature<dim> is of type
+ * TensorProductQuadrature<dim>.
+ */
+template <int dim>
+bool
+is_tensor_product(const Quadrature<dim> &quadrature);
```

Comment:
Can't you just write this as a member function for backward-compatibility similar to https://godbolt.org/z/W2VhZC or at least call this function.